include/alpaka/kernel/Traits.hpp

```diff
@@ -69,6 +70,25 @@ namespace alpaka
             }
         };
 
+        //! \brief The structure template to access to the functions attributes of a kernel function object.
+        //! \tparam TAcc The accelerator type
+        //! \tparam TKernelBundle The kernel object type, which includes the kernel function object and it's invocation
+        //! arguments.
+        template<typename TAcc, typename TKernelBundle>
+        struct FunctionAttributes
+        {
+            //! \param kernelBundle The kernel object instance, which includes the kernel function object and it's
+            //! invocation arguments.
+            //! \return KernelFunctionAttributes data structure instance. The default version always returns the
+            //! instance with fields which are set to zero.
+            ALPAKA_NO_HOST_ACC_WARNING
+            ALPAKA_FN_HOST_ACC static auto getFunctionAttributes([[maybe_unused]] TKernelBundle const& kernelBundle)
+                -> alpaka::KernelFunctionAttributes
+            {
+                return alpaka::KernelFunctionAttributes();
```

Comment:
Maybe this should `throw std::invalid_argument{...}` with a message about not being implemented for the given back-end, instead of returning a default-constructed object.

Reply:
done, thanks a lot.